File: DSA/LeetCode/Hashmaps/884. Uncommon Words from Two Sentences.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:

        dict1 = defaultdict(int)
        for w in s1.split():
            dict1[w] += 1

        for w in s2.split():
            dict1[w] += 1

        res = []
        for k, v in dict1.items():
            if v == 1:
                res.append(k)

        return res

File: DSA/LeetCode/Hashmaps/884. Uncommon Words from Two Sentences.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {string[]}
 */
var uncommonFromSentences = function (s1, s2) {

    let map1 = new Map()
    for (let w of s1.split(" ")) {
        map1.set(w, (map1.get(w) ?? 0) + 1);
    }


    for (let w of s2.split(" ")) {
        map1.set(w, (map1.get(w) ?? 0) + 1);
    }

    const uncommon = []
    for (let [word, value] of map1) {
        if (value == 1)
            uncommon.push(word)
    }

    return uncommon

};